internal/providers/docker/components/container/image.go

```diff
@@ -29,6 +29,12 @@ func (c *Container) ensureImage(ctx context.Context) error {
 		return fmt.Errorf("inspecting image: %w", err)
 	}
 	c.ImageID = inspection.ID
+
+	c.Command = make([]string, len(inspection.Config.Cmd))
+	for i, cmd := range inspection.Config.Cmd {
+		c.Command[i] = cmd
+	}
```

Comment:
Two problems with this:

1) Mutating the spec, not the state. Subsequent controller invocations won't have this information, so a `start` or `restart` operation (which doesn't call `ensureImage`) will not have access to the correct command.

2) This will shadow the docker compose file's command, which is the opposite of what we want, right? Presumably the compose file should shadow the code in the image.

Reply:
This is perhaps an aside but shouldn't both `start` and `restart`[0] call `ensureImage`? There's no guarantee the image will still be around if the container isn't running any more.